### PR TITLE
feat(KalturaAPI): services listeners-exposure and extra metrics

### PR DESCRIPTION
**Pull request by tweenn**
### Related Ticket(s)

[#3716 - [AnalyticsAPI] Add Video Progress and Video seek tag generation using the ibmstats.event function](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/3716)
[#7488 - Kaltura Player Metrics: For CTA clicks (& other events) - DDS Consulting](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/7488)

### Description

First I want to point out that there shouldn't be a breaking change in this PR - except by the error addressed in 34cc4387c83bd83c3883bfa29e800458fb4f3d49 (more on changelogs)

Second, this PR still need improvement and a review by a DBDM expert (more on changelogs)

Finally, this PR addresses 3 major changes in the kaltura-service - and analytics-video portion:

1. Make IBM Metrics firing optional - but turned on by default
2. Expose the readyCallback for custom manipulations of the player and their event-listeners
3. Add extra event listeners for IBM Metrics - like Mediacenter's cta-clicks and cta-impressions, and quartile and seek events from [#3716](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/3716)

It also has some unit-testing and coverage bumping, a bug fix in the analytics file (found while writing tests).

And I tried my best to make each commit be as semantic as possible with the code implemented - meaning the changelog may feel redundant.
 
### Changelog

* 3ac850d1ae768ec855915f3dc4c2efde5122c53a
Added namespaces to the player event-listeners.
This will keep IBM listeners in a safe position if a developer try to remove/unbind listeners, as like using a `kdp.removeListener('seek')` will remove all global/generic `'seek'` listeners, but will not remove the `'seek.ibm'` listener.

* ff7b5211cea4f4d38d8b8988e46edf4c74b2bd47
Added a param to `embedMedia` to make the IBM Metrics listeners optional - turned on by default
And fixed a documentation error where flashvars was set to boolean, when it was actually an object

* 34cc4387c83bd83c3883bfa29e800458fb4f3d49
Added a param to `embedMedia` that is a function that receives the `KDP` object in order to expose the `readyCallback`.

* ad9d2a1704aa21085e156964bcbdb819cc4dde92
The `AnalyticsAPI.videoPlayerStats` prop `executionPath` was expecting a `params.videoId`, that the old `VideoPlayer` service sent, but since the deprecation of that service in favor of `KalturaPlayer`, this param was changed/renamed to `mediaId` - and thus the metrics being fired right now are basically sending `{ executionPath: undefined }` to DBDM

* 58bc1417a762bb696685f83fe695f8fd780e174d
Added Mediacenter's CTA-CLICK listener and custom metric data to `KalturaPlayer` and the appropriate behaviour over the `AnalyticsAPI.videoPlayerStats` to add custom metrics if available.

* 3f4f3e70b88760772bca5a9c428d85c3008e89f3 & b6b45759d1dd98c491b97d8eaa34bb4c13aec976
Test and Coverage bumping - because I didn't like the values I saw 🤣 

* eb19d15d6ba6b6446ad4c8f2b85b86aa0bea726c
Created a draft for `seek` and `quartile` events.
Their data, like playerState and custom metrics are just a draft and totally need a DBDM expert review.

### CODE REVIEW

**Comment by annawen1** [comment]
Thank you @tweenn for this PR and adding so much detail in the description! One thing, can you make sure to sign the DCO? You can do this buy just adding the `I have read the DCO document and I hereby sign the DCO.` comment to this PR.

> DCO Assistant Lite bot: Thanks for your submission! We ask that you sign our Developer Certificate of Origin before we can accept your contribution. You can sign the DCO by adding a comment below using this text:
> 
> I have read the DCO document and I hereby sign the DCO.
>

**Comment by tweenn** [comment]
I have read the DCO document and I hereby sign the DCO.

File: packages/services/src/services/Analytics/Analytics.js [hunk 1]
```diff
@@ -213,7 +213,7 @@ class AnalyticsAPI {
    *       currentTime: 1,
    *       duration: 60,
    *       playerState: 1,
-   *       videoId: '1_9h94wo6b',
+   *       mediaId: '1_9h94wo6b',
```
Comment (jeffchew) [inline packages/services/src/services/Analytics/Analytics.js:216]: Is this technically an API change? Would probably need to note this in `breaking-changes.md` in the `services` package. (https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/master/packages/services/breaking-changes.md).

cc: @annawen1

File: packages/services/src/services/KalturaPlayer/KalturaPlayer.js [hunk 2]
```diff
@@ -234,17 +243,92 @@ class KalturaPlayerAPI {
...
-            kdp.addJsListener('playerPlayEnd', function() {
-              fireEvent({ playerState: 3, kdp, mediaId });
-            });
+            const kdp = document.getElementById(playerId);
+
+            if (useIbmMetrics) {
+              let startingSeekTime = 0;
+
+              kdp.addJsListener('playerPaused.ibm', () => {
+                fireEvent({ playerState: 1, kdp, mediaId });
```
Comment (jeffchew) [inline packages/services/src/services/KalturaPlayer/KalturaPlayer.js:252]: Not sure if we wanted to include as part of this PR or perhaps a separate PR, if I recall we were discussing also creating standard event listeners that any application could also subscribe to if they wanted to monitor these behaviors (e.g. append their own custom tracking triggers), which would be triggered by these Kaltura listeners. I'd be fine with adding this in a separate PR though.

**Comment by tweenn** [comment]
I'm closing this PR due to some problems with commit [#eb19d15](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/7652/commits/eb19d15d6ba6b6446ad4c8f2b85b86aa0bea726c)

Which is kinda solved in [Pull #7789](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/7789)
